llvm/test/CodeGen/PowerPC/aix-vec_insert_elt.ll

```diff
@@ -165,12 +165,12 @@ define <2 x i64> @testDoubleword(<2 x i64> %a, i64 %b, i64 %idx) {
 ;
 ; CHECK-32-LABEL: testDoubleword:
 ; CHECK-32:       # %bb.0: # %entry
-; CHECK-32-NEXT:    add 5, 6, 6
 ; CHECK-32-NEXT:    addi 7, 1, -32
+; CHECK-32-NEXT:    slwi 5, 6, 1
```

Comment:
This shows the advantage, after changing add to shift, on ppc,
- the shift can be combined with other shift, like rlwinm
- the shift lowest bit is 0, so ori can be used instead of addi(although this may not be an improvement, ori and addi have same latency)